src/stcal/resample/utils.py

```diff
@@ -0,0 +1,513 @@
+                # (or high read-out noise) single pixels. keep kernel size
+                # small to still give lower weight to extended CRs, etc.
+                ksz = weight_type[8 if selective_median else 7:]
+                if ksz:
+                    kernel_size = int(ksz)
+                    if not (kernel_size % 2):
+                        raise ValueError(
+                            'Kernel size of the median filter in IVM '
+                            'weighting must be an odd integer.'
+                        )
+                else:
+                    kernel_size = 3
+
+                ivm_copy = inv_variance.copy()
+
+                if selective_median:
+                    # apply median filter selectively only at
+                    # points of partially saturated sources:
+                    jumps = np.where(
+                        np.logical_and(dq & saturation, dqmask)
+                    )
+                    w2 = kernel_size // 2
+                    for r, c in zip(*jumps):
+                        x1 = max(0, c - w2)
+                        x2 = min(nx, c + w2 + 1)
+                        y1 = max(0, r - w2)
+                        y2 = min(ny, r + w2 + 1)
+                        data = ivm_copy[y1:y2, x1:x2][dqmask[y1:y2, x1:x2]]
+                        if data.size:
+                            inv_variance[r, c] = np.median(data)
+                        # else: leave it as is
+
+                else:
+                    # apply median to the entire inv-var array:
+                    inv_variance = median_filter(
+                        inv_variance,
+                        size=kernel_size
+                    )
+                bad_dqmask = np.logical_not(dqmask)
+                inv_variance[bad_dqmask] = ivm_copy[bad_dqmask]
+
+        else:
+            warnings.warn(
+                "var_rnoise array not available. "
+                "Setting drizzle weight map to 1",
+                RuntimeWarning
+            )
+            inv_variance = 1.0
+
+        weight = inv_variance * dqmask
+
+    elif weight_type == "exptime":
+        t, _ = get_tmeasure(model)
+        weight = np.full(data.shape, t)
+        weight *= dqmask
+
+    else:
+        weight = np.ones(data.shape, dtype=data.dtype) * dqmask
+
+    return weight.astype(np.float32)
+
+
+def get_tmeasure(model):
+    """
+    Check if the measurement_time keyword is present in the datamodel
+    for use in exptime weighting. If not, revert to using exposure_time.
+
+    Returns a tuple of (exptime, is_measurement_time)
+    """
+    try:
+        tmeasure = model["measurement_time"]
+    except KeyError:
+        return model["exposure_time"], False
+    if tmeasure is None:
+        return model["exposure_time"], False
+    else:
+        return tmeasure, True
+
+
+# FIXME: temporarily copied here to avoid this import:
+# from stdatamodels.jwst.library.basic_utils import bytes2human
+def bytes2human(n):
```

Comment:
This is unused right?

Reply:
yeah, a leftover since when we were checking for the available memory. Removed.